src/main/java/uk/ac/ebi/embl/api/validation/check/file/FileValidationCheck.java

```diff
@@ -211,6 +212,37 @@ protected int getAssemblyLevel(ValidationScope scope)
 		return assemblyLevel;
 	}
 
+	public void covid19SeqLengthCheck() throws ValidationEngineException {
+		if (sharedInfo.assemblyType != null && sharedInfo.assemblyType.equalsIgnoreCase(AssemblyType.COVID_19_OUTBREAK.getValue())) {
+			if (sharedInfo.chromosomeNames.size() > 1) {
+				throw new ValidationEngineException("COVID-19 genome submissions must have a single chromosome sequence", ReportErrorType.VALIDATION_ERROR);
+			}
+
+			long genomeSize = 0;
+			for (Map.Entry<String, AssemblySequenceInfo> entry : sharedInfo.sequenceInfo.entrySet()) {
+				AssemblySequenceInfo info = entry.getValue();
+
+				if (info.getAssemblyLevel() == 0 || info.getAssemblyLevel() == 1) {
+					if (!sharedInfo.agpEntryNames.contains(entry.getKey())) {
+						genomeSize += info.getSequenceLength();
```

Comment:
I think agpEntryName set contains object name of the agp row (first column, i.e name of the scaffold) not the component name(i.e could be contig name or another scaffold name), if so, it always excludes the scaffolds in length calculation.

if the contig is placed in agp then we should not include the contig length towards the genomeLength but we should include the scaffold length(unless it is placed on another scaffold) towards the genomeSize, this is my understanding

Reply:
- Consider unit tests for genome length computation. Optimally we would have test for: just contigs, just chromosomes, contigs + scaffolds, contigs + scaffolds + chromosomes, contigs + chromosomes.